lib/verification.js

```diff
@@ -23,17 +23,34 @@ export async function verifyRequest(rawBody, signature, key) {
 
 /** @type {import('..').FetchVerificationKeysInterface} */
 export async function fetchVerificationKeys(
-  { token = "", request = defaultRequest } = { request: defaultRequest }
+  { token = "", request = defaultRequest, cache = { id: "", keys: [] } } = { request: defaultRequest }
 ) {
-  const { data } = await request("GET /meta/public_keys/copilot_api", {
-    headers: token
+    try {
+    const headers = token
       ? {
           Authorization: `token ${token}`,
         }
-      : {},
-  });
+      : {};
+
+    if (cache.id) headers["If-None-Match"] = cache.id;
```

Comment:
Did you try if the changes work? I found the `etag` response and `If-None-Match` request header to be tricky in the past, there is some prefix that needs to be encoded ... just want to make sure we test the code in real-life before we merge the PR. If you run into any trouble let me know I can look what I did in other projects

Reply:
I was _just_ thinking about test running the changes. Will report back! 🫡

Reply:
Example script:

```js
import { request } from "@octokit/request";
import { fetchVerificationKeys } from "./index.js";

async function main() {
    const token = process.env.GITHUB_TOKEN;
    if (!token) throw new Error("GITHUB_TOKEN is not defined");

    const r = request.defaults({
        headers: {
            Authorization: `token ${token}`,
        }
    });

    const { id, keys } = await fetchVerificationKeys({ token });
    const cache = { id, keys };

    const firstRateLimitCheck = await r("GET /rate_limit");
    const firstRateLimitRemaining = firstRateLimitCheck.data.resources.core.remaining;
    console.log("Rate limit after fetching keys:", firstRateLimitRemaining);

    const response = await fetchVerificationKeys({ token, cache });

    const secondRateLimitCheck = await r("GET /rate_limit");
    const secondRateLimitRemaining = secondRateLimitCheck.data.resources.core.remaining;
    console.log("Rate limit after fetching keys from cache:", secondRateLimitRemaining);
}

main().catch(console.error);
```

Running it:

```js
@francisfuzz ➜ /workspaces/preview-sdk.js (9-cache) $ node example.js 
Rate limit after fetching keys: 4953
Rate limit after fetching keys from cache: 4953
```